streaming/sender.c

```diff
@@ -622,6 +622,15 @@ void *rrdpush_sender_thread(void *ptr) {
                 buffer_sprintf(s->build, "TIMESTAMP %ld", now);
                 sender_commit(s);
             }
+            if (s->version >= STREAM_VERSION_CLAIM) {
+                rrd_rdlock();
+                RRDHOST *h;
+                rrdhost_foreach_read(h) {
+                    if(h->claimed_id)
+                        rrdpush_claimed_id(h);
+                }
+                rrd_unlock();
+            }
```

Comment:
When an agent is streaming data from multiple hosts each one has a separate connection and sender.

Reply:
you are right done